scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1291,28 +1359,35 @@ class KubeletApi( object ):
         A class for querying the kubelet API
     """
 
-    def __init__( self, k8s, port=10255 ):
+    def __init__( self, k8s, port=10255, host_ip=None ):
         """
         @param k8s - a KubernetesApi object
         """
-        pod_name = k8s.get_pod_name()
-        pod = k8s.query_pod( k8s.namespace, pod_name )
-        spec = pod.get( 'spec', {} )
-        status = pod.get( 'status', {} )
-
-        host_ip = status.get( 'hostIP', None )
-
         if host_ip is None:
-            raise KubeletApiException( "Unable to get host IP for pod: %s/%s" % (k8s.namespace, pod_name) )
+            try:
+                pod_name = k8s.get_pod_name()
+                pod = k8s.query_pod( k8s.namespace, pod_name )
+                status = pod.get( 'status', {} )
+                host_ip = status.get( 'hostIP', None )
+                # Don't raise exception for now
+                # if host_ip is None:
+                #     raise KubeletApiException( "Unable to get host IP for pod: %s/%s" % (k8s.namespace, pod_name) )
+            except Exception:
+                global_log.exception( "couldn't get host ip" )
+                pass
 
         self._session = requests.Session()
         headers = {
             'Accept': 'application/json',
         }
         self._session.headers.update( headers )
 
-        self._http_host = "http://%s:%d" % ( host_ip, port )
-        self._timeout = 10.0
+        global_log.info('KubeletApi host ip = %s' % host_ip)
+        if host_ip:
+            self._http_host = "http://%s:%d" % ( host_ip, port )
+        else:
+            self._http_host = None
```

Comment:
Setting this to `None` has ramifications in later code e.g. [here](https://github.com/scalyr/scalyr-agent-2/blob/b6dbbd0ad76b2cd752cc8bea61ffebea146c95df/scalyr_agent/monitor_utils/k8s.py#L1395).  We need to make sure that if it's `None` then early abort future queries to the api.